### PR TITLE
Lotus58 glow

### PR DESCRIPTION
**Pull request by TweetyDaBird**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/tweetydabird/lotus58/atmeldfu/keymaps/default/info.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (waffle87) [inline keyboards/tweetydabird/lotus58/atmeldfu/keymaps/default/info.json:1]: Please remove the unnecessary whitespace from this file using `qmk format-json <file>`.

File: keyboards/tweetydabird/lotus58/config.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+// Bootmagic Lite key configuration (these do not transfer to Vial in info.json)
+#define BOOTMAGIC_LITE_ROW 0
+#define BOOTMAGIC_LITE_COLUMN 0
+#define BOOTMAGIC_LITE_ROW_RIGHT 5
+#define BOOTMAGIC_LITE_COLUMN_RIGHT 0
+
```
Comment (waffle87) [inline keyboards/tweetydabird/lotus58/config.h:11]:
```suggestion
```

File: keyboards/tweetydabird/lotus58/info.json [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            "right": {
+                "rotary": [
+                    {
+                        "pin_a": "F4",
+                        "pin_b": "F5",
+                        "resolution": 2
+                    }
+                ]
+            }
+        },
```
Comment (waffle87) [inline keyboards/tweetydabird/lotus58/info.json:68]:
```suggestion
        },
        "bootmagic": {
                "matrix": [5, 0]
        },
```

File: keyboards/tweetydabird/lotus58/info.json [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+		"tri_layer": true
+    },
+    "matrix_pins": {
+        "cols": ["B1", "B2", "B3", "B6", "F7", "F6"],
+        "rows": ["D4", "C6", "D7", "E6", "B4"]
+    },
+    "matrix_size": {
+        "cols": 6,
+        "rows": 10
+    },
```
Comment (waffle87) [inline keyboards/tweetydabird/lotus58/info.json:44]:
```suggestion
```

File: keyboards/tweetydabird/lotus58/lotus58.c [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "quantum.h"
+
+oled_rotation_t oled_init_kb(oled_rotation_t rotation) {
+	// Rotate OLED to fit
+    if (is_keyboard_left()) {
+        return OLED_ROTATION_270;
+	}
+    return OLED_ROTATION_270;
```
Comment (waffle87) [inline keyboards/tweetydabird/lotus58/lotus58.c:11]:
```suggestion

#ifdef OLED_ENABLE
oled_rotation_t oled_init_kb(oled_rotation_t rotation) {
        return OLED_ROTATION_270;
```

File: keyboards/tweetydabird/lotus58/lotus58.c [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+void render_logo(void) {
+	// QMK Logo
+    static const char PROGMEM qmk_logo[] = {
+        0x80, 0x81, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88, 0x89, 0x8A, 0x8B, 0x8C, 0x8D, 0x8E, 0x8F, 0x90, 0x91, 0x92, 0x93, 0x94,
+        0xA0, 0xA1, 0xA2, 0xA3, 0xA4, 0xA5, 0xA6, 0xA7, 0xA8, 0xA9, 0xAA, 0xAB, 0xAC, 0xAD, 0xAE, 0xAF, 0xB0, 0xB1, 0xB2, 0xB3, 0xB4,
+        0xC0, 0xC1, 0xC2, 0xC3, 0xC4, 0xC5, 0xC6, 0xC7, 0xC8, 0xC9, 0xCA, 0xCB, 0xCC, 0xCD, 0xCE, 0xCF, 0xD0, 0xD1, 0xD2, 0xD3, 0xD4, 0x00
+    };
+    oled_write_P(qmk_logo, false);
+}
```
Comment (waffle87) [inline keyboards/tweetydabird/lotus58/lotus58.c:22]:
```suggestion
}

bool oled_task_kb(void) {
        if (!oled_task_user()) {
                return false;
        }
        render_logo();
        return false;
}
#endif
```

File: keyboards/tweetydabird/lotus58/nanoboot/keymaps/default/info.json [hunk 7]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (waffle87) [inline keyboards/tweetydabird/lotus58/nanoboot/keymaps/default/info.json:1]: Please remove the unnecessary whitespace from this file using `qmk format-json <file>`.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/tweetydabird/lotus58/info.json (hunk 4 above)
Comment (TweetyDaBird) [reply keyboards/tweetydabird/lotus58/info.json:44]: Didn't even know this was redundant.

File: keyboards/tweetydabird/lotus58/config.h (hunk 2 above)
Comment (TweetyDaBird) [reply keyboards/tweetydabird/lotus58/config.h:11]: I'm leaving these for now, since there are others in there that doesn't work in the .JSON

File: keyboards/tweetydabird/lotus58/info.json (hunk 3 above)
Comment (TweetyDaBird) [reply keyboards/tweetydabird/lotus58/info.json:64]: I'm leaving these for now, since there are others in there that doesn't work in the .JSON

File: keyboards/tweetydabird/lotus58/config.h (hunk 2 above)
Comment (waffle87) [reply keyboards/tweetydabird/lotus58/config.h:11]: What do you mean they don't work? They auto generated defines from `info.json` work fine from my testing. Or do you meaning something like Vial repository has not adopted these data-driven options..

Comment (TweetyDaBird) [reply keyboards/tweetydabird/lotus58/config.h:11]: Since the other lines below Aare things not migrated to data driven and these do indeed not work in Vial as data driven (yet), I feel that leaving them all until ALL are data driven should be OK. At that point I'll happily migrate it all.

Comment (zvecr) [reply keyboards/tweetydabird/lotus58/config.h:11]: These items should be migrated to info.json, as bootmagic is a feature which is has been covered by data driven for 2 develop cycles.

Unfortunately the whole "vial compatibility" reasoning is non of this repos concern.

File: keyboards/tweetydabird/lotus58/nanoboot/keymaps/default/info.json [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
```
Comment (zvecr) [inline keyboards/tweetydabird/lotus58/nanoboot/keymaps/default/info.json:1]: `info.json` files at the keymap level wont work, so none of this configuration is being picked up.

File: keyboards/tweetydabird/lotus58/info.json [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+    "matrix_pins": {
+        "cols": ["B1", "B2", "B3", "B6", "F7", "F6"],
+        "rows": ["D4", "C6", "D7", "E6", "B4"]
+    },
+    "pin_compatible": "promicro",
+    "processor": "atmega32u4",
+    "rgblight": {
+        "brightness_steps": 17,
+        "hue_steps": 17,
```
Comment (zvecr) [inline keyboards/tweetydabird/lotus58/info.json:45]:
```suggestion
```

File: keyboards/tweetydabird/lotus58/info.json [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+        "led_count": 70,
+        "pin": "D3",
+        "saturation_steps": 17,
```
Comment (zvecr) [inline keyboards/tweetydabird/lotus58/info.json:48]:
```suggestion
```

File: keyboards/tweetydabird/lotus58/info.json [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+        "grave_esc": false,
+        "magic": false,
+        "mousekey": false,
+        "nkro": true,
+        "oled": true,
+        "rgblight": true,
+        "space_cadet": false,
+        "split": true,
+        "tap_dance": false,
+		"tri_layer": true
```
Comment (zvecr) [inline keyboards/tweetydabird/lotus58/info.json:35]:
```suggestion
        "bootmagic": true,
        "command": false,
        "console": false,
        "encoder": true,
        "extrakey": true,
        "grave_esc": false,
        "magic": false,
        "mousekey": true,
        "nkro": true,
        "oled": true,
        "rgblight": true,
        "space_cadet": false,
        "split": true,
		"tri_layer": true
```

Space permitting, the feature-set should be more "default" like

```suggestion
        "bootmagic": true,
        "command": false,
        "console": false,
        "encoder": true,
        "extrakey": true,
        "mousekey": true,
        "nkro": true,
        "oled": true,
        "rgblight": true,
        "split": true,
		"tri_layer": true
```

**Review by zvecr — COMMENTED** [review]
Keymap folder per revision seems a bit unnecessary given its the same PCB in all cases.

```
keyboards/tweetydabird/lotus58/keymaps/default/keymap.c
keyboards/tweetydabird/lotus58/atmeldfu/rules.mk
keyboards/tweetydabird/lotus58/caterina/rules.mk
keyboards/tweetydabird/lotus58/nanoboot/rules.mk
```

Using some existing terminology could also help with consistency;
* caterina -> promicro
* atmeldfu -> elite_c

As a decent proportion of user will have no idea when to use `caterina` vs `atmeldfu`.

File: keyboards/tweetydabird/lotus58/info.json (hunk 3 above)
Comment (TweetyDaBird) [reply keyboards/tweetydabird/lotus58/info.json:59]: This is missing the part for the main hand. Editing it manually.

File: keyboards/tweetydabird/lotus58/info.json (hunk 9 above)
Comment (zvecr) [reply keyboards/tweetydabird/lotus58/info.json:43]: Because some of these are the defaults, because other add little value, because you were asked to so that this PR can be submitted to the repo.

Comment (TweetyDaBird) [reply keyboards/tweetydabird/lotus58/info.json:43]: Either of the first two reasons had been quite sufficient if they had been given to start with. The third statement is not a reason, but an opinion.

Comment (TweetyDaBird) [reply keyboards/tweetydabird/lotus58/info.json]: I didn't argue an opinion. I asked the reason why, since I genuinely didn't know why you wanted them removed, since they wouldn't in any way be wrong. But if they simply mimimc what has become default since I added them a very long time ago, they should indeed be removed, and I did so once you gave the reason.

Yes, I have admittedly 'argued' or more discussed on other points where my opinion differed. If that is not allowed, feel free to point to that rule. Otherwise I will continue to ask questions if I don't understand the reasoning. Or in this case the reason wasn't given. It might be obvious to you, it just wasn't to me. 

No slight intended. Just a question.

**Comment by TweetyDaBird** [comment]
I believe all requested changes (manual edits) should be fixed and tested now. Let me know if there is more that needs fixing.

**Comment by TweetyDaBird** [comment]
Ah, right. Then I misunderstood didn't read carefully enough. Now, I'll have to think on that. This sort of creates an issue going to Vial.

Again trying to have a discussion, not arguing/being a PITA more than necessary. I agree that this isn't your issue to solve, I'm just trying to find the middle ground where it's acceptable here, and doesn't totally shoot me in the foot there, creating more work as a whole. Feel free to just ignore me if you don't want to have the discussion. 

Moving the key-maps to a central folder doesn't work in Vial as the key-maps there have to be dependent on the boot-loader (lack of space) and the standard for the Vial key-map is a single folder with the name Vial.

Do you have any good suggestions for a workaround? 

Can the boot-loader assignment move to key-map level? It actually can can't it? (Although I'm not sure if that the correct thing to do).

**Comment by zvecr** [comment]
> where it's acceptable here, and doesn't totally shoot me in the foot there

The only thing that is of concern here if its acceptable to QMK. Optimising, even if its only in part, for Vial is zero on the scale.

However it should be noted that having a common folder for keymaps does not stop at a later date the introduction of 

```
keyboards/tweetydabird/lotus58/atmeldfu/keymaps/something/keymap.c
keyboards/tweetydabird/lotus58/caterina/keymaps/something/keymap.c
```

As long as "something" doesnt also exist within `keyboards/tweetydabird/lotus58/keymaps`.

You can also use have conditional blocks within the compilation units, with `#ifdef BOOTLOADER_CATERINA` and things like `post_rules.mk` to disable features completely.

**Comment by TweetyDaBird** [comment]
> The only thing that is of concern here if its acceptable to QMK. Optimising, even if its only in part, for Vial is zero on the scale.

I fully understand the position. However, had I just swallowed the first suggestion and not questioned, it would probably resulted in another PR with an update fairly soon, and then another, and another... Making us both batty...

The reason I asked however is this:

> You can also use have conditional blocks within the compilation units, with `#ifdef BOOTLOADER_CATERINA` and things like `post_rules.mk` to disable features completely.

That should have been obvious to me, having used the '#ifdef OLED' etc... It wasn't for some reason. So, from the bottom of my heart, thank you for making me feel damned stupid... That was going beyond...

It might perhaps take me a while to iron it out to my satisfaction (and yours), but I think you just gave me the key...

**Comment by drashna** [comment]
This is the fifth PR for this board: #20281, #18554, #15867, #12594.

**Comment by TweetyDaBird** [comment]
> This is the fifth PR for this board: #20281, #18554, #15867, #12594.

Yeah. I don't do good with GitHub time limits and such. ADHD. Either things happen fast, and I say focused, or I loose focus and hyper focus on something else, and end up procrastinating. Was sort of why I managed to annoy Joel before(justified from his side Btw). I was more trying to keep myself on task than anything else.  

And git isn't my strong suit to start with. Is there a way to re-open a PR from my side? I would have preferred that.

**Comment by TweetyDaBird** [comment]
Well, a work day worth of testing. It work as is, and armed with the above tip, I feel I can take this and make Vial work without shooting myself in the foot. I'm happy with it. Let me know if there is anything else that needs fixed.

File: keyboards/tweetydabird/lotus58/elite_c/rules.mk [hunk 12]
```diff
@@ -0,0 +1 @@
+BOOTLOADER = atmel-dfu
```
Comment (elpekenin) [inline keyboards/tweetydabird/lotus58/elite_c/rules.mk:1]: Should have a `info.json` instead of `rules.mk`, on this same folder, to setup the bootloader

File: keyboards/tweetydabird/lotus58/info.json [hunk 13]
```diff
@@ -0,0 +1,148 @@
+{
+    "manufacturer": "Tweetys Wild Thinking",
+    "keyboard_name": "Lotus 58 Glow (QMK)",
+    "maintainer": "TweetyDaBird",
+    "bootloader_instructions": "Short marked pads on PCB, or hold top-outer key when plugging in each hand.",
+	"bootmagic": {
+        "matrix": [0, 0]
+    },
```
Comment (elpekenin) [inline keyboards/tweetydabird/lotus58/info.json:8]: Default value, not needed
```suggestion
```

File: keyboards/tweetydabird/lotus58/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+
+bool oled_task_user(void) {
+	// Render the OLED
+	if (is_keyboard_left()) {
+		print_status_narrow();
+		//render_logo();
+    } else {
+		print_status_narrow();
+        //render_logo();
+    }
```
Comment (elpekenin) [inline keyboards/tweetydabird/lotus58/keymaps/default/keymap.c:89]: Could be reduced into
```suggestion
		print_status_narrow();
```

File: keyboards/tweetydabird/lotus58/nanoboot/rules.mk [hunk 15]
```diff
@@ -0,0 +1,2 @@
+BOOTLOADER = qmk-hid
```
Comment (elpekenin) [inline keyboards/tweetydabird/lotus58/nanoboot/rules.mk:1]: Could be on json too (the bootloader part)

File: keyboards/tweetydabird/lotus58/promicro/rules.mk [hunk 16]
```diff
@@ -0,0 +1 @@
+BOOTLOADER = caterina
```
Comment (elpekenin) [inline keyboards/tweetydabird/lotus58/promicro/rules.mk:1]: JSON again

**Review by elpekenin — CHANGES_REQUESTED** [review]

File: keyboards/tweetydabird/lotus58/nanoboot/rules.mk (hunk 15 above)
Comment (TweetyDaBird) [reply keyboards/tweetydabird/lotus58/nanoboot/rules.mk:1]: Respectfully, but no.

Once the Bootloader size is able to be ported over to info.json, I will happily change this. Until then, I categorically refuse.
I understand the reasoning that info.json is the superior choice, and what we are striving too.

But in this case, with two that intertwined options in two different files in two different formats, it just causes confusion. You change them as a whole, or you do not change them. So this one stays for now.

**Review by drashna — APPROVED** [review]

File: keyboards/tweetydabird/lotus58/elite_c/rules.mk (hunk 12 above)
Comment (zvecr) [reply keyboards/tweetydabird/lotus58/elite_c/rules.mk]: Sorry but these suggestions are completely incorrect, as then it does not become a build target.

File: keyboards/tweetydabird/lotus58/promicro/rules.mk (hunk 16 above)
Comment (zvecr) [reply keyboards/tweetydabird/lotus58/promicro/rules.mk]: Same as <https://github.com/qmk/qmk_firmware/pull/20386#discussion_r1167577864> here too.

**Review by zvecr — APPROVED** [review]
Ive fixed up things after the bad info you were given, as well as a few minor tidy ups. All good to go from my front now.

**Review by keyboard-magpie — APPROVED** [review]
